### ISSUE TITLE
Render all fields, not just those rendered on the page

### ISSUE THREAD

**Issue by balinterdi** [cardstack/cardstack#300]
Models might have fields that we also want to render in the sidebar in Edit mode even if they were not explicitly marked for editing in the template (by wrapping them in a `cs-field` or a `{{content.field}}`).

Marking a field as `sticky` (other name suggestions: `pinned`, `editorial`) means that field is always rendered for editing, no matter if it shows up on the page. Care should be taken not to render such fields twice (which can be the case if it's also rendered explicitly in the template).

**Comment by ef4** [comment]
This is indeed the solution we discussed, though I am beginning to wonder if we should just always have a section in the sidebar to show _all_ the fields that are present but weren't rendered in the template.

Perhaps all we really need is an intelligent way to order and/or organize the fields, so that the ones relevant to the template appear first, but you can still find the others further down.

**Comment by balinterdi** [comment]
I think that would make sense though I wonder what the original intent behind only showing the ones rendered on the page was. For example,

>  if we should just always have a section in the sidebar to show all the fields

would mean that we show all related fields of all associated records in "owned" relationships. If a resource has n related resources and each related resource has m fields, that would mean n*m fields in the sidebar which can be pretty overwhelming (cluttered). In the case of documents and chapters, that would mean all chapter titles and bodies are listed (not to mention images which I think would be another owned relationship).

However, we have to provide a way to edit fields in owned relationships as by their definition they cannot be edited in their content-type (as they don't have an isolated view).

I think two concepts are currently coupled which should not be: the inclusion of a field in the editor sidebar and rendering that field in the page. One idea to decouple them and solve the above problem is to introduce a new CS helper/component that would only include the fields in the editor, without rendering them, something like:

```handlebars
{{#cs-editor-fields-for content as |documentFields|}}
  {{documentFields 'thumbnail'}}
  {{documentFields 'headerBackgroundImage'}}
  {{documentFields 'attachments'}}
{{/cs-editor-fields}}

{{#cs-editor-fields-for selectedChapter as |chapterFields|}}
  {{chapterFields 'readingTime'}}
{{/cs-editorFields}}
```

We could also just wrap them in a div that we set to `display: none` but that would feel hacky, and we might be able to add other functionality in the `cs-editor-fields-for` helper.

**Comment by ef4** [comment]
You're right that the fully-owned relationships make the right-edge editing more complex, but I think that would be true for any of these solutions. I don't see an obvious way to automatically discover which owned child resources should have their editorial fields included in the sidebar at any given time.

I think a `cs-editor-fields-for` component is plausible, but it feels to me like not truly a concern of the component.

I don't think we should over-curate the right edge. We already have one system for determining which fields a user can see or edit (the grants), and if they have permission, there's no reason they shouldn't be able to find those fields in the right edge, regardless of what fields a particular card is using.

**Comment by ef4** [comment]
An idea that Chris and I have discussed is to allow an optional customized right-edge component that is provided by the card to give a prettier, card-specific UI, but even in that case it's really important that the user can flip between the custom UI and the raw field editors. The idea being: it's their data, they may want to do things with it that the card making didn't intend, as long as there's no grant or constraint forbidding something it's all inspectable and directly editable.

**Comment by balinterdi** [comment]
Ok, I like this – so I can go ahead with showing all fields of the resource-under-edit and all fields in all related owned records, right?

**Comment by ef4** [comment]
Yes, if you see a way to design the right edge to make the "all fields in related owned records" reasonably nice. We may want to get some design help.

**Comment by balinterdi** [comment]
What about splitting this into two phases and rendering all fields of the resource-under-edit in Phase 1 and then adding all fields of all related owned records in Phase 2 (then the design help would only be needed for Phase 2)?

**Comment by ef4** [comment]
Sure.

**Comment by balinterdi** [comment]
Closed by #408
